components/configuration-mgt/org.wso2.carbon.identity.configuration.mgt.core/src/main/java/org/wso2/carbon/identity/configuration/mgt/core/constant/SQLConstants.java

```diff
@@ -72,6 +90,28 @@
             "(NAME), " +
             "LAST_MODIFIED = VALUES(LAST_MODIFIED), HAS_FILE = VALUES(HAS_FILE), HAS_ATTRIBUTE = VALUES" +
             "(HAS_ATTRIBUTE), TYPE_ID = VALUES(TYPE_ID)";
+    public static final String INSERT_OR_UPDATE_RESOURCE_POSTGRESQL = "INSERT INTO IDN_CONFIG_RESOURCE" +
+            "(ID,TENANT_ID,NAME, CREATED_TIME,LAST_MODIFIED,HAS_FILE, HAS_ATTRIBUTE,TYPE_ID) " +
+            "VALUES(?, ?, ?, ?, ?, ?, ?, ?) ON CONFLICT (ID) DO UPDATE " +
+            "SET TENANT_ID = EXCLUDED.TENANT_ID, NAME = EXCLUDED.NAME, LAST_MODIFIED = EXCLUDED.LAST_MODIFIED, " +
+            "HAS_FILE =EXCLUDED.HAS_FILE, HAS_ATTRIBUTE = EXCLUDED.HAS_ATTRIBUTE, TYPE_ID = EXCLUDED.TYPE_ID";
+    public static final String INSERT_OR_UPDATE_RESOURCE_DB2 = "MERGE INTO IDN_CONFIG_RESOURCE USING " +
+            "\"SYSIBM\".DUAL ON (ID= ? ) WHEN MATCHED THEN UPDATE SET TENANT_ID = ?, NAME = ?, LAST_MODIFIED = ?, " +
```

Comment:
`SYSIBM`: Is this a general constant in DB2?

Reply:
Otherwise, say this represents the default admin user. Then it may not work in all the deployments.

Reply:
The query used in mssql is valid for db2 also. "SYSIBM" is not using anymore. Changed in 5a0504e